functions.php

```diff
@@ -91,17 +104,30 @@ function as_schedule_single_action( $timestamp, $hook, $args = array(), $group =
 		return is_int( $pre ) ? $pre : 0;
 	}
 
-	return ActionScheduler::factory()->create(
-		array(
-			'type'      => 'single',
-			'hook'      => $hook,
-			'arguments' => $args,
-			'when'      => $timestamp,
-			'group'     => $group,
-			'unique'    => $unique,
-			'priority'  => $priority,
-		)
-	);
+	try {
+		return ActionScheduler::factory()->create(
+			array(
+				'type'      => 'single',
+				'hook'      => $hook,
+				'arguments' => $args,
+				'when'      => $timestamp,
+				'group'     => $group,
+				'unique'    => $unique,
+				'priority'  => $priority,
+			)
+		);
+	} catch ( Exception $exception ) {
+		ActionScheduler::logger()->log(
+			0,
+			sprintf(
+				/* translators: %1$s is the name of the hook to be enqueued, %2$s is the exception message. */
+				__( 'Caught exception while enqueuing action "%1$s": %2$s', 'action-scheduler' ),
+				$hook,
+				$exception->getMessage()
+			)
+		);
+		return 0;
```

Comment:
It was already possible for these functions to return zero, but I wonder if we should take this opportunity to highlight this via the docblock? Something like:

```diff
- * @return int The action ID.
+ * @return int The action ID. If zero, the action was not created and an error may have been logged in the database.
```